cmd/kubectl-testkube/commands/dashboard.go

```diff
@@ -19,53 +22,69 @@ func NewDashboardCmd() *cobra.Command {
 	)
 
 	cmd := &cobra.Command{
-		Use:   "dashboard",
-		Short: "Open testkube dashboard",
-		Long:  `Open testkube dashboard`,
+		Use:     "dashboard",
+		Aliases: []string{"d"},
+		Short:   "Open testkube dashboard",
+		Long:    `Open testkube dashboard`,
 		Run: func(cmd *cobra.Command, args []string) {
-
-			ui.Verbose = true
 			ui.Logo()
 
 			uri := fmt.Sprintf("http://localhost:%d", DashboardLocalPort)
 			if useGlobalDashboard {
 				uri = DashboardURI
 			}
 
-			dashboardAddress := fmt.Sprintf("%s/apiEndpoint?apiEndpoint=localhost:%d/%s", uri, ApiServerPort, ApiVersion)
-			ui.Success("The dashboard is accessible here:", dashboardAddress)
-			ui.Success("Port forwarding is started for the test results endpoint, hit Ctrl+C to stop")
+			apiURI := fmt.Sprintf("localhost:%d/%s", ApiServerPort, ApiVersion)
+			dashboardAddress := fmt.Sprintf("%s/apiEndpoint?apiEndpoint=%s", uri, apiURI)
+			apiAddress := fmt.Sprintf("http://%s", apiURI)
 
 			var (
-				wg      sync.WaitGroup
-				command *exec.Cmd
-				err     error
+				commandsToKill []*exec.Cmd
+				err            error
 			)
 
+			// kill background port-forwarded processes
+			defer func() {
+				for _, command := range commandsToKill {
+					if command != nil {
+						err := command.Process.Kill()
+						ui.ExitOnError("killing command: "+command.String(), err)
+					}
+				}
+			}()
+
+			// if not global dasboard - we'll try to port-forward current cluster API
 			if !useGlobalDashboard {
-				wg.Add(1)
-				go func() {
-					defer wg.Done()
-					command, err = forwardKubernetesPort(true, namespace, DashboardName, DashboardLocalPort, DashboardPort)
-				}()
-
-				wg.Wait()
+				command, err := asyncPortForward(namespace, DashboardName, DashboardLocalPort, DashboardPort)
 				ui.ExitOnError("port forwarding dashboard endpoint", err)
+				commandsToKill = append(commandsToKill, command)
 			}
 
+			command, err := asyncPortForward(namespace, ApiServerName, ApiServerPort, ApiServerPort)
+			ui.PrintOnError("port forwarding api endpoint", err)
+			commandsToKill = append(commandsToKill, command)
+
+			// check for api and dasboard to be ready
+			readinessCheck(apiAddress, dashboardAddress)
```

Comment:
I would either check method returns or didn't return anything from this methid

Reply:
simplified it a little